Source/engine/points_in_rectangle_range.hpp

```diff
@@ -0,0 +1,355 @@
+#pragma once
+
+#include <iterator>
+
+#include "point.hpp"
+#include "rectangle.hpp"
+
+namespace devilution {
+class PointsInRectangleIteratorBase {
+protected:
+	PointsInRectangleIteratorBase(Point origin, int majorDimension, int majorIndex, int minorIndex)
+	    : origin(origin)
+	    , majorDimension(majorDimension)
+	    , majorIndex(majorIndex)
+	    , minorIndex(minorIndex)
+	{
+	}
+
+	PointsInRectangleIteratorBase(Point origin, int majorDimension, int index = 0)
+	    : PointsInRectangleIteratorBase(origin, majorDimension, index / majorDimension, index % majorDimension)
+	{
+	}
+
+public:
+	using iterator_category = std::random_access_iterator_tag;
+	using difference_type = int;
+	using value_type = Point;
+	using pointer = const value_type *;
+	using reference = value_type &&;
+
+	virtual reference operator*() const = 0;
```

Comment:
Let's avoid virtual methods, they're slower to call. Simply remove this line, remove `override` throughout.

Reply:
I've split this into a separate branch for now while I try work out the syntax required to get it building on GCC. The only reason I had this virtual method was to make the base class abstract, but that's not necessary given the base class doesn't fully define any of the required concepts for an iterator.